python/cudf/cudf/core/dataframe.py

```diff
@@ -1864,8 +1856,7 @@ def _get_columns_by_label(self, labels, downcast=False):
         )
         return out
 
-    @annotate("DATAFRAME_BINARYOP", color="blue", domain="cudf_python")
-    def _binaryop(
+    def _prep_for_binop(
```

Comment:
Can we add a return type annotation, if not a docstring, for this function?

Reply:
Ditto below for `_binaryop`.

Reply:
I'd like to postpone that to the subsequent PR if that's all right. I'm not yet sure exactly what these APIs are going to look like once I merge this logic with what we have for index types.

Reply:
No problemo